src/edu/psu/sweng500/team8/gui/BoardGUI.java

```diff
@@ -0,0 +1,261 @@
+import javax.swing.JTextField;
+
+import edu.psu.sweng500.team8.coreDataStructures.Cell;
+import edu.psu.sweng500.team8.coreDataStructures.CellCoordinates;
+import edu.psu.sweng500.team8.play.GameSession;
+
+public class BoardGUI extends JPanel {
+	
+	private static final int BOARD_SIZE = 486;
+	private BlockGUI[][] blocks;
+	private GameSession gameSession;
+	private CellGUI selectedCell;
+	private Set<CellGUI> highlightedIncorrectCells = new HashSet<CellGUI>();
+	
+	public BoardGUI() {
+		
+		setPreferredSize(new Dimension(BOARD_SIZE, BOARD_SIZE));
+		setMaximumSize(new Dimension(BOARD_SIZE, BOARD_SIZE));
+		setMinimumSize(new Dimension(BOARD_SIZE, BOARD_SIZE));
+
+		blocks = new BlockGUI[3][3];
+
+		GridBagConstraints gridBagConstraints;
+		setBorder(javax.swing.BorderFactory
+				.createLineBorder(new java.awt.Color(0, 0, 0)));
+		setLayout(new java.awt.GridBagLayout());
+
+		blocks[0][0] = new BlockGUI();
+		gridBagConstraints = new GridBagConstraints();
+		gridBagConstraints.gridx = 0;
+		gridBagConstraints.gridy = 0;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 0);
+		add(blocks[0][0], gridBagConstraints);
+
+		blocks[0][1] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 0;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 0);
+		add(blocks[0][1], gridBagConstraints);
+
+		blocks[0][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 0;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 0);
+		add(blocks[0][2], gridBagConstraints);
+
+		blocks[1][0] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 0;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 1, 0, 0);
+		add(blocks[1][0], gridBagConstraints);
+
+		blocks[1][1] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+		add(blocks[1][1], gridBagConstraints);
+
+		blocks[1][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 1);
+		add(blocks[1][2], gridBagConstraints);
+
+		blocks[2][0] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 0;

```

Comment:
If I understand the API correctly, GridBagConstraints.gridx represent
column position (as in a two-dimensional graph) while a two-dimensional
array's first bracket usually represent rows. As for a two-dimensional
array, it doesn't really matter how we use it as long as we keep it
consistent, but that's the convention.
(http://www.homeandlearn.co.uk/java/multi-dimensional_arrays.html)

On Sun, Jul 12, 2015 at 10:08 PM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/BoardGUI.java
> <https://github.com/garnryang/Team8/pull/13#discussion_r34429545>:
>
> > +		gridBagConstraints.gridx = 1;
> > +		gridBagConstraints.gridy = 1;
> > +		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
> > +		add(blocks[1][1], gridBagConstraints);
> > +
> > +		blocks[1][2] = new BlockGUI();
> > +		gridBagConstraints = new java.awt.GridBagConstraints();
> > +		gridBagConstraints.gridx = 2;
> > +		gridBagConstraints.gridy = 1;
> > +		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
> > +//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 1);
> > +		add(blocks[1][2], gridBagConstraints);
> > +
> > +		blocks[2][0] = new BlockGUI();
> > +		gridBagConstraints = new java.awt.GridBagConstraints();
> > +		gridBagConstraints.gridx = 0;
>
> Why are the x/y indices reversed from the row/column?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/13/files#r34429545>.
>